rdmo/projects/viewsets.py

```diff
@@ -529,52 +532,109 @@ def get_queryset(self):
+        try:
+            copy_value_id = int(request.data.pop('copy_set_value'))
+        except KeyError as e:
+            raise ValidationError({
+                'copy_set_value': [_('This field may not be blank.')]
+            }) from e
+        except ValueError as e:
+            raise NotFound from e
+
+        # look for this value in the database, using the users permissions, and
         # collect all values for this set and all descendants
-        currentValues = self.get_queryset().filter_set(currentValue)
+        try:
+            copy_value = Value.objects.filter_user(self.request.user).get(id=copy_value_id)
+            copy_values = Value.objects.filter_user(self.request.user).filter_set(copy_value)
+        except Value.DoesNotExist as e:
+            raise NotFound from e
 
-        # de-serialize the posted new set value and save it, use the ValueSerializer
-        # instead of ProjectValueSerializer, since the latter does not include project
-        set_value_serializer = ValueSerializer(data={
-            'project': parent_lookup_project,
-            **request.data
-        })
-        set_value_serializer.is_valid(raise_exception=True)
-        set_value = set_value_serializer.save()
-        set_value_data = set_value_serializer.data
+        # init list of values to return
+        response_values = []
+
+        set_value_id = request.data.get('id')
+        if set_value_id:
+            # if an id is given in the post request, this is an import
+            try:
+                # look for the set value for the set we want to import into
+                set_value = Value.objects.filter_user(self.request.user).get(id=set_value_id)
+
+                # collect all non-empty values for this set and all descendants and convert
+                # them to a list to compare them later to the new values
+                set_values = Value.objects.filter_user(self.request.user).filter_set(set_value)
+                set_values_list = set_values.exclude_empty().values_list('attribute', 'set_prefix', 'set_index')
+                set_empty_values_list = set_values.filter_empty().values_list(
+                    'attribute', 'set_prefix', 'set_index', 'collection_index'
+                )
+            except Value.DoesNotExist as e:
+                raise NotFound from e
+        else:
+            # otherwise, we want to create a new set and need to create a new set value
+            # de-serialize the posted new set value and save it, use the ValueSerializer
+            # instead of ProjectValueSerializer, since the latter does not include project
+            set_value_serializer = ValueSerializer(data={
+                'project': parent_lookup_project,
+                **request.data
+            })
+            set_value_serializer.is_valid(raise_exception=True)
+            set_value = set_value_serializer.save()
+            set_values_list = set_empty_values_list = []
```

Comment:
does it also need a `set_values = ...` here? otherwise later it can be undefined

Reply:
Yes, while in principle, `set_values` will not be accessed since `somethingsomething in set_empty_values_list` is checked first, it is cleaner to add `set_values = Value.objects.none()`.